app/src/main/java/com/orange/ods/app/ui/MainTopAppBar.kt

```diff
@@ -63,37 +62,34 @@ fun MainTopAppBar(
     val showSearchAction = topAppBarState.titleRes.value == R.string.navigation_item_search
 
     val title = stringResource(id = topAppBarState.titleRes.value)
-    val navigationIcon: (@Composable () -> Unit)? = if (shouldShowUpNavigationIcon && topAppBarState.isNavigationIconEnabled) {
-        {
-            Icon(
-                imageVector = Icons.Filled.ArrowBack,
-                contentDescription = stringResource(id = R.string.top_app_bar_back_icon_desc)
-            )
-        }
-    } else null
+    val navigationIcon = if (shouldShowUpNavigationIcon && topAppBarState.isNavigationIconEnabled) {
+        OdsTopAppBarNavigationIcon(Icons.Filled.ArrowBack, stringResource(id = R.string.top_app_bar_back_icon_desc), upPress)
+    } else {
+        null
+    }
 
-    val actions: @Composable RowScope.() -> Unit = {
-        if (showSearchAction) {
-            TopAppBarSearchAction(searchedText = topAppBarState.searchedText)
-        } else {
-            TopAppBarActions(topAppBarState, onSearchActionClick) { changeThemeDialogVisible = true }
-        }
+    val actions = if (showSearchAction) {
+        listOf(getTopAppBarSearchTextFieldAction(searchedText = topAppBarState.searchedText))
+    } else {
+        getTopAppBarActions(topAppBarState, onSearchActionClick) { changeThemeDialogVisible = true }
     }
 
+    val overflowMenuActions = getTopAppBarOverflowMenuActions(topAppBarState)
+
     if (topAppBarState.isLarge) {
-        OdsLargeTopAppBar(
+        OdsLargeTopAppBarInternal(
```

Comment:
So we have to implement the `OdsSearchTopAppBar` quickly to avoid to call these "Internal" functions.